### PR TITLE
Pallet staking events to named enum

### PR DESCRIPTION
**Pull request by Szegoo**
Closes: #12226

Polkadot address: 126X27SbhrV19mBFawys3ovkyBS87SGfYwtwa8J2FjHrtbmA

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: frame/nomination-pools/test-staking/src/lib.rs [hunk 1]
```diff
@@ -41,7 +41,10 @@ fn pool_lifecycle_e2e() {
 		// have the pool nominate.
 		assert_ok!(Pools::nominate(RuntimeOrigin::signed(10), 1, vec![1, 2, 3]));
 
-		assert_eq!(staking_events_since_last_call(), vec![StakingEvent::Bonded(POOL1_BONDED, 50),]);
+		assert_eq!(
+			staking_events_since_last_call(),
+			vec![StakingEvent::Bonded { stash: POOL1_BONDED, amount: 50 },]
```
Comment (kianenigma) [inline frame/nomination-pools/test-staking/src/lib.rs:46]:
```suggestion
			vec![StakingEvent::Bonded { stash: POOL1_BONDED, amount: 50 }]
```

**Review by kianenigma — APPROVED** [review]

**Comment by kianenigma** [comment]
please add your polkadot address to all your PRs @Szegoo :D

**Comment by kianenigma** [comment]
@Ank4n feel free to merge once you review.

**Comment by kianenigma** [comment]
bot rebase

**Review by Ank4n — APPROVED** [review]

**Pull request merged**